### PR TITLE
Tweak roxygen2 workflow

### LINKED ISSUE #927: Change default NAMESPACE
To not export anything — makes it a little more consistent with documenting after the first time you've added an `@export`

### LINKED ISSUE #963: Should check() run document on first call?
Currently, if you create a package and then immediately check it, it does not automatically run `document()`. But once you've run `document()` once, `check()` will automatically run it for you. This seems mildly confusing and worth reconsidering.

Probably related to #927

### PR DESCRIPTION
**Pull request by hadley**
* New explicit roxygen argument to create_package(), use_description(), and use_namespace()

* Defaults have been tweaked so that devtools::check() will run document() the first time you call it, and functions will only be exported after you've explicitly exported them

Fixes #927. Fixes #963

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 # usethis (development version)
 
+* `create_package()` gains a `roxygen` argument. If `TRUE` (the default), 
+  adds a `RoxygenNote` field to the `DESCRIPTION` (which means the first run
+  of `devtools::check()` will re-document the package, #963), and creates an 
+  empty `NAMESPACE` (which means you'll need always need explicit `@export`
```
Comment (jennybc) [inline NEWS.md]:
```suggestion
  empty `NAMESPACE` (which means you'll always need an explicit `@export` 
```

File: R/create.R [hunk 2]
```diff
@@ -9,9 +9,10 @@
 #' Both functions can be called on an existing project; you will be asked
 #' before any existing files are changed.
 #'
+#' @inheritParams use_description
 #' @param path A path. If it exists, it is used. If it does not exist, it is
 #'   created, provided that the parent path exists.
-#' @inheritParams use_description
+#' @param roxygen Are you planning on using roxygen to document your package?
```
Comment (jennybc) [inline R/create.R]:
```suggestion
#' @param roxygen Do you plan to use roxygen2 to document your package?
```

File: R/namespace.R [hunk 3]
```diff
@@ -1,13 +1,21 @@
 #' Use a basic `NAMESPACE`
 #'
-#' This `NAMESPACE` exports everything, except functions that start
-#' with a `.`.
+#' If `roxygen` is `TRUE` generates a empty `NAMESPACE` that exports nothing;
```
Comment (jennybc) [inline R/namespace.R]:
```suggestion
#' If `roxygen` is `TRUE` generates an empty `NAMESPACE` that exports nothing;
```

File: R/namespace.R [hunk 4]
```diff
@@ -1,13 +1,21 @@
...
+#' you'll need to explicitly export functions with `@export`. If `roxygen`
+#' is `FALSE`, generates a default `NAMESPACE` that exports all functions
+#' except those that start with `.`.
 #'
+#' @param roxygen Are you planning to use roxygen2?
```
Comment (jennybc) [inline R/namespace.R]:
```suggestion
#' @param roxygen Do you plan to manage `NAMESPACE` with roxygen2?
```

File: tests/testthat/test-use-roxygen.R [hunk 5]
```diff
@@ -11,19 +11,3 @@ test_that("use_roxygen_md() adds DESCRIPTION fields to naive package", {
...
-      scoped_temporary_package()
-      cat(
-        "RoxygenNote: 6.0.1\n",
-        file = proj_path("DESCRIPTION"),
-        append = TRUE
-      )
-      expect_error_free(use_roxygen_md())
-    }
-  )
-})
```
Comment (jennybc) [inline tests/testthat/test-use-roxygen.R]: Is there a reason to delete this / delete this in this PR?

File: R/namespace.R [hunk 6]
```diff
@@ -1,13 +1,21 @@
...
 #' @seealso The [namespace chapter](https://r-pkgs.org/namespace.html) of
 #'   [R Packages](https://r-pkgs.org).
-#'
 #' @export
-use_namespace <- function() {
+use_namespace <- function(roxygen = TRUE) {
   check_is_package("use_namespace()")
-  use_template("NAMESPACE")
```
Comment (jennybc) [inline R/namespace.R]: Should we also be deleting the `NAMESPACE` template file then?

File: tests/testthat/test-use-roxygen.R (hunk 5 above)
Comment (jennybc) [reply tests/testthat/test-use-roxygen.R]: Related: should we add some tests around the new features?

File: R/namespace.R (hunk 6 above)
Comment (hadley) [reply R/namespace.R]: Oooh yes

File: tests/testthat/test-use-roxygen.R (hunk 5 above)
Comment (hadley) [reply tests/testthat/test-use-roxygen.R]: It didn't really apply any more. I'll have a play with some tests to see if they feel worthwhile.

Comment (hadley) [reply tests/testthat/test-use-roxygen.R]: I had try, but they're mostly going to be golden tests, so I don't think they're particularly important for this change.

**Pull request merged**